vscode-trace-extension/src/extension.ts

```diff
@@ -59,6 +60,17 @@ export function activate(context: vscode.ExtensionContext): ExternalAPI {
         if (e.affectsConfiguration('trace-compass.traceserver.url') || e.affectsConfiguration('trace-compass.traceserver.apiPath')) {
             updateTspClient();
         }
+
+        if (e.affectsConfiguration('trace-compass.traceserver.url')) {
+            const newUrl = getTraceServerUrl();
```

Comment:
Investigating remote deployments of this plugin, I've been doing some checking on VSCode utilities for Remote deployments and found the `asExternalUri` function in the [`env`](https://code.visualstudio.com/api/references/vscode-api#env) namespace. Basically, in a desktop deployment, that function does nothing, because `external` is the same as local, but in a remote deployment, that function transforms the URI into a form that the client can use to access a client running on the host and also does any port forwarding, if the application is set up for that. You can see it in action in the ['Live Preview'](https://github.com/microsoft/vscode-livepreview/blob/d37c5d583a53db5add2ac476c3d30c4b3c13da95/src/connectionInfo/connection.ts#L74-L75) plugin, which, like the Trace Extension, opens up a server on the remote machine on certain ports. With that in mind, perhaps the output of `getTraceServerUrl` should be passed through that utility to ensure that it's accessible.

The caveat that I can see is that if a user is using a VSCode deployment that does not implement the necessary infrastructure, it might be desirable to let the user specify the exact shape of the URL they want the views to use so that the user can set up access to the service manually.

Reply:
Hi @colin-grant-work, thank you for the investigation. Are you OK to merge this PR and discuss this later if necessary?